### DISCUSSION TITLE
Clearing up of old status checks

### DISCUSSION THREAD

**Discussion by lapitsky** [community/community#25190] [Actions]
<p>I’m playing with the GitHub actions now.</p>
<p>The first piece of logic that I want to introduce is a validation of our pull request messages.</p>
<p>I’ve implemented the action and it correctly checks PR messages and fails status checks when needed. The problem I’m facing now is how to recover from these failed checks.</p>
<p>When I fix the PR messages the check turns green and this new check is just added to the list of all checks, failed and succeeded. I see smth like “2 failing and 3 successful checks”. What I want is to enable merge when the last check succeeds.</p>
<p>Could you please help me or point me to the relevant documentation?</p>

**Comment by gimenete** [comment, marked as answer]
<p>Hi!</p>
<p>If you make you generate a successful status check with the same name, it should “overwrite” the existing, failing one.</p>

**Comment by lapitsky** [comment]
<p>Thank you for the reply!</p>
<p>I think that I’m missing some important concept. I’ve created a JS action that checks PR messages and if certain criteria are not met it calls <code>core.setFailed</code>. The action is used from a workflow.</p>
<p>I could not see another explicit function like <code>core.setSucceeded</code>. I also do not see how I can pass a check name somewhere.</p>
<p>Thanks in advance</p>

**Comment by DylanVann** [comment]
<p>I also do not see an API to generate a successful check.</p>
<p>This should not be marked as solved.</p>

**Comment by RandellDawson** [comment]
<p>Did you ever figure out a way to do this?  I am stuck with the same problem.</p>
